### ISSUE TITLE
Report deliverable id for each encountered problem when publishing a DITA OT project file

### ISSUE THREAD

**Issue by raducoravu** [dita-ot/dita-ot#3948]
## Description
I publish a DITA OT project file, various problems are reported in the console, how do I know for which deliverable I obtain which problem? 

## Possible Solution
Maybe besides the [DOT...] problem ID we should also add a [deliverable.id] in the reported message string.

## Potential Alternatives
Not sure if there are any.

**Comment by xephon2** [comment]
One example could be when a keyref points to a missing key because of a profiling filter or a missing key definition map.

**Comment by raducoravu** [comment]
As an example, let's say you publish a DITA OT project file with lots of deliverables, each with a context and profiling setting. And you receive this problem in the console:

     [DOTJ048I] Unable to find key definition for key reference X

You have no idea for what context (map + filters) this problem was encountered.

**Comment by jelovirt** [comment]
It would be difficult to add the deliverable as a context to each message. Would it be enough if we have a info log message that "Processing delivarable {}" and you would need to search the log to find which deliverable it belongs to? At least for 3.7.x.

**Comment by raducoravu** [comment]
@jelovirt yes this would help and would probably be better than adding extra information to each message.

**Comment by chrispy-snps** [comment]
@jelovirt - that would work perfectly fine for us too.

**Comment by chrispy-snps** [comment]
An additional thought as I watch the messages for 60+ deliverables scroll by... it would be useful if the message could indicate the current and total deliverable numbers, something like:

```
Publishing deliverable del-pdf-final-dcug (3 of 60)
```
